### PR TITLE
overlord/devicestate: keep track of tried recovery system

### PR DESCRIPTION
**Pull request by bboozzoo**
Record a recovery system that was tried.

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -1048,6 +1051,72 @@ func (m *DeviceManager) ensureSeedInConfig() error {
...
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	deviceCtx, err := DeviceCtx(m.state, nil, nil)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	outcome, label, err := boot.InspectTryRecoverySystemOutcome(deviceCtx)
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go:1088]: now that we are looking at the full circle: we went -> recover -> back to run mode here,  this helper should check that label belongs as well to modeenv CurrentRecoverySystems and not report success unless that's the case I think but is not doing that atm afaict.

For the rest this code seems correct to

Comment (anonymouse64) [reply overlord/devicestate/devicemgr.go:1088]: yeah agreed, this doesn't seem to be doing that, I can try to quickly push a separate patch enabling that behavior

Comment (anonymouse64) [reply overlord/devicestate/devicemgr.go:1094]: I pushed a patch for this here, hopefully it's in the right direction to unblock this

**Review by anonymouse64 — APPROVED** [review]
the bits I didn't touch lgtm

**Comment by pedronis** [comment]
my concern has been addresses, unblocking

**Review by pedronis — APPROVED** [review]
thanks @bboozzoo and @anonymouse64

**Comment by bboozzoo** [comment]
Thanks for the reviews and patches!

**Pull request merged**